src/internal/Subscription.ts

```diff
@@ -75,6 +75,14 @@ export class Subscription implements SubscriptionLike {
     }
 
     if (isFunction(_unsubscribe)) {
+      // It's not possible to null the _unsubscribe member as there are many
+      // classes that are derived from Subscriber (which derives from
+      // Subscription) that implement an _unsubscribe method as a mechanism for
+      // obtaining unsubscription notifications and some of those subscribers
+      // are recycled. Deleting the member will release the reference to any
+      // teardown functions passed in the constructor and will leave any
+      // methods intact.
+      delete (this as any)._unsubscribe;
```

Comment:
Won't this cause a de-opt in V8 though, i.e. put the entire `Subscription` instance in dictionary mode?

Reply:
Yeah. We should really implement an `_unsubscribe` method that's always present and store the teardown that's passed to the ctor in a property that's actually writable.

Reply:
A quick fix could also be to make the constructor just call `this.add(teardown)` instead of assigning it to `_unsubscribe`

Reply:
Yeah. I considered that, but, IIRC, the implementation of `add` creates another `Subscription` - passing the teardown to the ctor.

Reply:
I wasn't too happy about creating a closure in the ctor, either. I think we should just bit the bullet and add the method.

Reply:
Sounds good!

Reply:
Given that the `Subscription` implementation does so much other clean up, I suppose it make some sense to release the `_unsubscribe` teardown, too, but - I'm curious - is there some particular use case in which you want to continue to hold the subscription after you've called unsubscribe? I mean, if it's inside - i.e. was added to - another longer-lived subscription, the unsubscription of the shorter-lived subscription should remove it from the longer-lived one, so I'm wondering about the use case and whether or not there's a bug someplace else.

Reply:
I have a fairly large codebase that uses web workers and `MessagePort`s, which are notorious for needing manual memory management. This code is also running inside a browser extension in a background page (i.e. a long-living context) and will create new ports very often for new tabs, proxying objects within that tab, etc. Given these attributes, I want to do everything in my power to make sure there is no memory leaking. If `_unsubscribe` is not nulled out, I can't guarantee that, because it is very easy to accidentally break out of the nice parent-cleaning tree and there's no way to ensure this at scale. For example, somewhere in the code someone may have written
```ts
subscription.add(() => otherSubscription.unsubscribe())
```
instead of
```ts
subscription.add(otherSubscription)
```
and the nice parent-cleaning chain would be broken, and the parent would hold on to the callback. This is a simple example, but this could also happen with more layers in between, e.g. a class that has an internal `Subscription` bag, but exposes an `unsubscribe` or `close` method itself, that is then added to a `Subscription`. It should be possible to rely on at least the low-level to clean things up properly, so that we may hold a reference to an empty  `Subscription` longer than needed, but at least the underlying `MessagePort` and heavier resources like the web worker etc are cleaned up.